### PR TITLE
[KEYCLOAK-3334] - NodeJS Policy Enforcer

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Comment by abstractj** [comment]
@pedroigor great work Pedro. Now we need to add integration tests for this. See: https://github.com/keycloak/keycloak-nodejs-connect/tree/master/test

**Comment by pedroigor** [comment]
@abstractj Add `test/keycloak-connect-rest-enforcer-spec.js`. Also fixed code style, `make` should not log errors.

I'm running the new test as follows:

```shell
node test/keycloak-connect-rest-enforcer-spec.js
```

**Comment by pedroigor** [comment]
Btw, one of the new tests require keycloak server from upstream. Reason being we fixed an issue when evaluating permissions for scopes.

**Comment by abstractj** [comment]
@pedroigor good morning Pedro. Do you think this could be the reason of some tests failing on Travis?

**Comment by pedroigor** [comment]
@abstractj, yes. A single test is failing because of this. This one `# Should test no access to protected resource and scope view and delete.`

File: test/keycloak-connect-rest-enforcer-spec.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+'use strict';
+
+const admin = require('./utils/realm');
+const NodeApp = require('./fixtures/node-console/index').NodeApp;
+
+const test = require('blue-tape');
+const roi = require('roi');
+const getToken = require('./utils/token');
+
+const realmName = 'service-node-realm';
```
Comment (abstractj) [inline test/keycloak-connect-rest-enforcer-spec.js]: @pedroigor I tried to test it with Keycloak server from master. But I believe that's not the reason why tests are failing.

There's a duplicate with https://github.com/keycloak/keycloak-nodejs-connect/blob/master/test/keycloak-connect-rest-spec.js#L26. That's an easy fix, just rename to something like `policy-enforcer-realm`.

Even after renaming it, I get 3 tests failing. You can reproduce it by doing the following steps:

1. clone/build Keycloak from master
2. Start the server
```
#!/bin/bash

set -v

# Start the server
$KEYCLOAK/bin/add-user-keycloak.sh -u admin -p admin

# Start the server
$KEYCLOAK/bin/standalone.sh -Djava.net.preferIPv4Stack=true \
                            -Dkeycloak.migration.action=import \
                            -Dkeycloak.migration.provider=singleFile \
                            -Dkeycloak.migration.file=./test/fixtures/auth-utils/nodejs-test-realm.json \
                            -Dkeycloak.migration.strategy=OVERWRITE_EXISTING > keycloak.log 2>&1 &
``` 
3. Run `make tests`

Comment (pedroigor) [reply test/keycloak-connect-rest-enforcer-spec.js]: Let me try it out. Thanks.

Comment (pedroigor) [reply test/keycloak-connect-rest-enforcer-spec.js]: I'm able now to get `make tests` running. But I had to change `keycloak-connect-rest-enforcer-spec.js` to also pass the `realmName` when calling `getToken({ realmName })`.

**Comment by abstractj** [comment]
This PR depends on https://github.com/keycloak/keycloak-nodejs-connect/pull/143. After the merge we can rebase and I believe tests will pass.

**Comment by wtrocki** [comment]
Absolutely love that change. Discovered that by accident! 
Was waiting for this for long time! Great work.

Is there documentation for this?

**Comment by MumblesNZ** [comment]
@abstractj was any progress made on the documentation?

**Comment by abstractj** [comment]
@MumblesNZ for questions, please ask on the user mailing list.

**Comment by pedroigor** [comment]
@MumblesNZ, we have the `keycloak.enforcer` method documented here https://github.com/keycloak/keycloak-nodejs-connect/blob/master/index.js#L191. Please let us know if you think we should provide more details.

**Comment by pedroigor** [comment]
@MumblesNZ, I've sent a PR to official docs too https://github.com/keycloak/keycloak-documentation/pull/654.

**Pull request merged**